src/main/webapp/assets/script.js

```diff
@@ -222,16 +225,92 @@ function initMap() {
 
   // Get all posts on the page and show them as markers.
   posts.forEach((post) => {
+    const width = getMapMarkerIconSize(post.numOfPeopleFoodWillFeed, 'width');
+    const height = getMapMarkerIconSize(post.numOfPeopleFoodWillFeed, 'height');
+    const icon = {
+      url: getMapMarkerIconUrl(post.eventStartTime),
+      scaledSize: new google.maps.Size(width, height),
+      origin: new google.maps.Point(0, 0),
+      anchor: new google.maps.Point(0, 0),
+    };
+
     new google.maps.Marker({
       position: {lat: post.location.lat, lng: post.location.long},
       map: map,
       title: post.organizationName,
+      opacity: getMapMarkerOpacity(post.eventStartTime, post.eventEndTime),
+      icon: icon,
     });
   });
 
 /* eslint-enable no-undef */
 }
 
+/**
+ * Calculates the opacity of a given map marker.
+ * @param {Date} eventStartTime - The start time of the event.
+ * @param {Date} eventEndTime - The end time of the event.
+ * @return {number} - The opacity to show the marker as.
+ */
+function getMapMarkerOpacity(eventStartTime, eventEndTime) {
+  const now = new Date();
+  const startTime = eventStartTime;
+  const endTime = eventEndTime;
+
+  // If the date is in the future, our marker should have full opacity.
+  if (startTime > now) {
+    return 1;
+  }
+
+  // If the event is over, our marker should not be on the map at all.
+  if (endTime < now) {
+    return 0;
+  }
+
+  // Else, we need to map the full range of 1 to 0 to the range of time
+  // between eventStartTime and eventEndTime.
+  const adjusted = 1 - ((now - startTime) / (endTime - startTime));
+  return adjusted;
+}
+
+/**
+ * Returns the appropriate marker icon for a marker.
+ * Grey if the event hasn't started yet, red otherwise.
+ * @param {Date} eventStartTime - The start time of the event.
+ * @return {string} - The location of the marker icon.
+ */
+function getMapMarkerIconUrl(eventStartTime) {
+  const now = new Date();
+
+  // If the date is in the future, return the grey marker.
+  if (eventStartTime > now) {
+    return './assets/greymarker.png';
+  }
+
+  // Else, a red marker.
+  return './assets/redmarker.png';
+}
+
+/**
+ * Returns the appropriate marker size for a single dimension of a marker, based on the number
+ * of people the food will feed. Uses a logistic function so that incredibly large and small
+ * numbers still have reasonable icon sizes.
+ * @param {number} numOfPeopleFoodWillFeed - The number of people the event's food will feed.
+ * @param {string} dimensionType - The dimension type, either 'width' or 'height'.
+ * @return {number} - The marker dimension.
+ */
+function getMapMarkerIconSize(numOfPeopleFoodWillFeed, dimensionType) {
+  // Sanity check input for invalid values.
+  if (dimensionType !== 'width' && dimensionType !== 'height') {
+    return null;
+  }
+
+  // Apply the logistic function to the input.
+  const bounds = dimensionType === 'width' ? [28, 70] : [45, 113];
```

Comment:
Make the numbers constants. I would recommend having a separate method for the logistic implementation but that can be a TODO.

Reply:
Sounds good! It should be quick to move the logistic function into another method, so I'll just do that as well :)